tests/fsharp/FSharpSuite.Tests.fsproj

```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>net472;$(FSharpNetCoreProductTargetFramework)</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' == 'Unix'">$(FSharpNetCoreProductTargetFramework)</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' == 'Unix' or '$(BUILDING_USING_DOTNET)' == 'true'">$(FSharpNetCoreProductTargetFramework)</TargetFrameworks>
```

Comment:
I do not get it - why do we need to remove net472 if BUILDING_USING_DOTNET ?

Reply:
I don't think net472 works with BUILDING_USING_DOTNET. At least I couldn't make it work.

Reply:
I think it was because project dependencies also remove it, so the build would fail otherwise.

Reply:
> The alternative is to run the tests with
> 
> `dotnet test .\tests\fsharp --framework:net9.0 --filter "nameof-fsi"`

The thing is this won't work because it will try to build the net472 target anyway, but will not get compatible dependencies for it. For example FSharp.Test.Utilities  build net9.0 only with BUILDING_USING_DOTNET.

Reply:
Yeah, I actually always run
`dotnet build FSharp.Compiler.Service.sln`
and then
`dotnet test .\tests\fsharp --no-build --framework:net9.0 --filter "nameof-fsi"`